### PR TITLE
fix ODS no feature generation bug

### PR DESCRIPTION
**Pull request by SophieHYe**
In this commit, some bugs are fixed:1) null pointer exceptions and 2)P4J features for multiple changed files.

### CODE REVIEW

**Comment by martinezmatias** [comment]
Hi @SophieHYe 
Thanks a lot for the bug fixing.
Do you have a test cases with the case that exposes the bug and assert that is not present any more?
Thanks
Matias

**Comment by SophieHYe** [comment]
Hi @martinezmatias,

Thanks for reminding. I have updated the existing test case, where the previous result of Defects4J Math-42 human patch was UNKNOWN due to the bug of feature extraction. Since the bug has fixed, the result now for this patch has updated to CORRECT.

File: src/main/java/fr/inria/coming/codefeatures/FeatureAnalyzer.java [hunk 1]
```diff
@@ -97,14 +93,18 @@ public AnalysisResult analyze(IRevision revision, RevisionResult previousResults
...
-							jsonFeature.add("P4J",p4jfeatures.getFeatures());
-						}
+						
 						changesArray.add(jsonFeature);
 					}
 				}
 				
 			}
+			
+			FeaturesResult p4jfeatures = (FeaturesResult) new P4JFeatureAnalyzer().analyze(revision, nameFile+"");
```
Comment (martinezmatias) [inline src/main/java/fr/inria/coming/codefeatures/FeatureAnalyzer.java:103]: Why appends an empty string? Is it missing something (eg. file extension)?

File: src/main/java/fr/inria/prophet4j/utility/CodeDiffer.java [hunk 2]
```diff
@@ -168,7 +168,7 @@ private RepairGenerator newRepairGenerator(DiffEntry diffEntry) {
                 if (insertOperation instanceof InsertOperation) {
                     dstNode = insertOperation.getSrcNode(); // ...
                 }
-            } else if (deleteOperation != null) {
+            } else if (deleteOperation != null && DEL != null && insertOperation == null ) { // && DEL != null
```
Comment (martinezmatias) [inline src/main/java/fr/inria/prophet4j/utility/CodeDiffer.java:171]: let's remove the comment //

**Comment by martinezmatias** [comment]
Thanks @SophieHYe, added just two small comments.

File: src/main/java/fr/inria/prophet4j/utility/CodeDiffer.java (hunk 2 above)
Comment (SophieHYe) [reply src/main/java/fr/inria/prophet4j/utility/CodeDiffer.java]: Sure, just removed!

File: src/main/java/fr/inria/coming/codefeatures/FeatureAnalyzer.java (hunk 1 above)
Comment (SophieHYe) [reply src/main/java/fr/inria/coming/codefeatures/FeatureAnalyzer.java]: just convert from object type to String type, updated to nameFile.toString()

**Comment by martinezmatias** [comment]
great @SophieHYe !

**Pull request merged**